### ISSUE TITLE
Integrate progress bars for Async requests

### ISSUE THREAD

**Issue by sxthimons** [ropensci/crul#168]
Currently, it looks like Async does not allow for the progress to be displayed when making requests. This would be helpful to determine if a request is still in progress or failing.

**Comment by sckott** [comment]
You can pass `httr::progress()` to the `progress` param in https://docs.ropensci.org/crul/reference/HttpRequest.html which are used in https://docs.ropensci.org/crul/reference/AsyncVaried.html 

But I doubt that's what you are looking for. I assume you are looking for progress across all the requests rather than within each request?

**Comment by sxthimons** [comment]
I'd prefer across each request. When using the `Async$new(urls = urls, progress = httr::progress())` it doesn't look like it's accepting it trying to create the initial R6 class variable: 
`Error in initialize(...) :  unused argument (progress = list(NULL, NULL, NULL, NULL, list(FALSE, function (down, up)...`

I've also tried to pass it through the `opts` argument on the Async as well.

**Comment by sckott** [comment]
I don't think this will happen unless you want to add it yourself via a pull request
